File: app/src/main/java/tpcreative/co/qrscanner/common/entities/HistoryDao.java
package tpcreative.co.qrscanner.common.entities;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;
import java.util.List;

@Dao
public interface HistoryDao {
    @Insert
    void insert(HistoryEntity... history);

    @Update
    void update(HistoryEntity... history);

    @Delete
    void delete(HistoryEntity... history);

    @Query("DELETE FROM history")
    public void deleteAllItems();

    @Query("Select * FROM history ORDER BY updatedDateTime DESC")
    List<HistoryEntity> loadAll();

    @Query("Select * FROM history ORDER BY id DESC LIMIT 10")
    List<HistoryEntity> loadLatestItems();

    @Query("Select * FROM history WHERE contentUnique = :contentUnique")
    HistoryEntity loadItem(String contentUnique);

}

File: app/src/main/java/tpcreative/co/qrscanner/common/entities/HistoryDao.kt
package tpcreative.co.qrscanner.common.entities
import androidx.room.*

@Dao
interface HistoryDao {
    @Insert
    fun insert(vararg history: HistoryEntity?)
    @Update
    fun update(vararg history: HistoryEntity?)
    @Delete
    fun delete(vararg history: HistoryEntity?)
    @Query("DELETE FROM history")
    fun deleteAllItems()
    @Query("DELETE FROM history where uuId =:uuId")
    fun deleteSpecific(uuId: String?)
    @Query("Select * FROM history ORDER BY updatedDateTime DESC")
    fun loadAll(): MutableList<HistoryEntity>?
    @Query("Select * FROM history where isSynced =:isSynced ORDER BY updatedDateTime DESC")
    fun loadAll(isSynced: Boolean): MutableList<HistoryEntity>?
    @Query("Select * FROM history WHERE contentUnique = :contentUnique")
    fun loadItem(contentUnique: String?): HistoryEntity?
    @Query("Select * FROM history WHERE id = :id")
    fun loadItem(id: Int?): HistoryEntity?
    @Query("Select * FROM history WHERE favorite = :isFavorite")
    fun loadAllItem(isFavorite : Boolean): MutableList<HistoryEntity>?
    @Query("Select * FROM history WHERE uuId = :uuId")
    fun loadUUId(uuId: String?): HistoryEntity?
}